share/spack/gitlab/cloud_pipelines/stacks/python/spack.yaml

```diff
@@ -0,0 +1,1834 @@
+    # - py-virtualenv-clone
+    # - py-virtualenvwrapper
+    # - py-visdom
+    # - py-voluptuous
+    # - py-vsc-base
+    # - py-vsc-install
+    # - py-vsts
+    # - py-vsts-cd-manager
+    # - py-walinuxagent
+    # - py-wand
+    # - py-wandb
+    # - py-warlock
+    # - py-warpx
+    # - py-wasabi
+    # - py-watchdog
+    # - py-wcsaxes
+    # - py-wcwidth
+    # - py-weave
+    # - py-webencodings
+    # - py-webkit-server
+    # - py-weblogo
+    # - py-webob
+    # - py-websocket-client
+    # - py-websockets
+    # - py-werkzeug
+    # - py-wget
+    # - py-whatshap
+    # - py-wheel
+    # - py-whichcraft
+    # - py-whoosh
+    # - py-widgetsnbextension
+    # - py-wincertstore
+    # - py-word2number
+    # - py-workload-automation
+    # - py-wradlib
+    # - py-wrapt
+    # - py-wstool
+    # - py-wub
+    # - py-wurlitzer
+    # - py-wxmplot
+    # - py-wxpython
+    # - py-xarray
+    # - py-xattr
+    # - py-xdot
+    # - py-xenv
+    # - py-xgboost
+    # - py-xlrd
+    # - py-xlsxwriter
+    # - py-xlwt
+    # - py-xmlrunner
+    # - py-xmltodict
+    # - py-xopen
+    # - py-xpyb
+    # - py-xvfbwrapper
+    # - py-xxhash
+    # - py-yacs
+    # - py-yahmm
+    # - py-yajl
+    # - py-yamlreader
+    # - py-yapf
+    # - py-yarl
+    # - py-yolk3k
+    # - py-youtube-dl
+    # - py-yq
+    # - py-yt
+    # - py-ytopt
+    # - py-ytopt-autotune
+    # - py-zarr
+    # - py-zc-buildout
+    # - py-zc-lockfile
+    # - py-zict
+    # - py-zipp
+    # - py-zope-event
+    # - py-zope-interface
+    # - py-zxcvbn
+
+    # - pythia8+python
+    # - qscintilla+python
+    # - rdkit
+    # - reditools
+    # - redland-bindings
+    # - reprimand+python
+    # - saga-gis+python
+    # - scons
+    # - sensei+python
+    # - sgpp+python
+    # - simsipm+python
+    # - simulationio
+    # - sirius+python
+    # - snakemake
+    # - sollve+python
+    # - sumo
+    # - sz+python
+    # - talloc
+    # - tasmanian+python
+    # - tfel+python_bindings
+    # - thrift+python
+    # - timemory+python
+    # - tskit
+    # - tulip
+    # - uqtk+pyuqtk
+    # - verrou
+    # - vigra+python
+    # - vtk+python
+    # - xcfun
+    # - xrootd+python
+    # - xtensor-python
+    # - z3+python
+
+  mirrors: { "mirror": "s3://spack-binaries-develop/python" }
+
+  gitlab-ci:
+
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd ${SPACK_CONCRETE_ENV_DIR}
+      - spack env activate --without-view .
+      - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+      - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+      - spack -d ci rebuild > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
+
+    mappings:
+      - match: ['@:']
+        runner-attributes:
+          tags: ["spack", "public", "large", "x86_64"]
+
```

Comment:
@adamjstewart Without an `image` in this runner mapping, it looks like gitlab is picking one at random (or I just don't understand how it chooses in that case).  The jobs that failed with the error you pasted above seem to have picked `busybox` as the image, which doesn't have the shell support needed.  If you add:

```
image: { "name": "ghcr.io/spack/e4s-ubuntu-18.04:v2021-10-18", "entrypoint": [""] }
```

and nest it at the same indentation level as `tags` here, I think that should do the trick.  See the other stacks for examples too, e.g. e4s [here](https://github.com/spack/spack/blob/01d077d4bc3aac21b2f8f83f6fa7071ea18418e5/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml#L223).